go/vt/sqlparser/sql.y

```diff
@@ -5714,7 +5731,7 @@ convert_type:
   {
     $$ = &ConvertType{Type: string($1), Length: $2}
   }
-| CHAR length_opt charset_opt
+| CHAR length_opt charset_convert_opt
```

Comment:
This now uses the separate explicit option that has the conversions documented at https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html